From: Sven Eckelmann <sven@narfation.org>
Date: Tue, 11 Oct 2022 16:45:20 +0200
Subject: Revert "batman-adv: Drop duplicated dev_{hold,put} compat helpers"

This is still necessary for QCAs 5.4 kernel with QCAs mac80211.

This reverts commit a2fe1d413176d5856a7a3f3e736208a6872a2b85.

diff --git a/batman-adv/src/compat-hacks.h b/batman-adv/src/compat-hacks.h
index 99a1bb39fedcbd2069a0f790a624fdeaed12b47c..ddd86710c38b0d3a6b85783739fb327e474548ab 100644
--- a/compat-hacks.h
+++ b/compat-hacks.h
@@ -94,6 +94,24 @@ br_multicast_has_router_adjacent(struct net_device *dev, int proto)
 
 #if LINUX_VERSION_IS_LESS(5, 15, 0)
 
+static inline void batadv_dev_put(struct net_device *dev)
+{
+	if (!dev)
+		return;
+
+	dev_put(dev);
+}
+#define dev_put batadv_dev_put
+
+static inline void batadv_dev_hold(struct net_device *dev)
+{
+	if (!dev)
+		return;
+
+	dev_hold(dev);
+}
+#define dev_hold batadv_dev_hold
+
 static inline void batadv_eth_hw_addr_set(struct net_device *dev,
 					  const u8 *addr)
 {
